### DISCUSSION TITLE
Understanding ArtifactHub

### DISCUSSION THREAD

**Discussion by octonawish-akcodes** [artifacthub/hub#2792] [Q&A]
Hi community, I am Abhishek currently a student. I am trying to understand an issue on showing some policies on artifacthub from a github repo and ive gone through how gatekeeper implemented this. I want to know how can I test this out on my own like for ex if ive a cloned repo and I want to try implementing by adding some test pol to show on AH how can I do that. can you help. I looked up other implementation and saw that there is should be repo and pkg yamls in the repo but I am a bit confused how to get started can you guide me

**Comment by tegioz** [comment]
Hi @octonawish-akcodes 👋

The best starting point would be the [repositories guide](https://artifacthub.io/docs/topics/repositories/). There you can find a detailed guide for the specific repository kind you'd like to add. As you'll see in the guides, you can structure the AH metadata in multiple ways depending on your needs (single package, multiple packages, versioning or not, etc).

Once your repo is ready, you can test it out locally or in the [staging environment](https://staging.artifacthub.io). The latter has the advantage that it doesn't require you to setup anything, but it'll process your repository every 30 minutes, so if you need to run multiple tests it'll be slower. You can also launch your own Artifact Hub instance locally by using the provided [Helm chart](https://artifacthub.io/packages/helm/artifact-hub/artifact-hub). Another alternative would be to use the [Gitpod environment](https://gitpod.io/#https://github.com/artifacthub/hub).

Additionally, you can check how other projects are doing this for inspiration:

- Keda: <https://github.com/kedacore/external-scalers/tree/main/artifacthub>
- Keptn: <https://github.com/keptn-contrib/artifacthub>
- Kubewarden: <https://github.com/kubewarden/allowed-fsgroups-psp-policy>
- Tinkerbell: <https://github.com/tinkerbell/hub/tree/artifacthub-manifests/actions>
- Gatekeeper: <https://github.com/open-policy-agent/gatekeeper-library/tree/master/artifacthub>

Hope this helps 🙂

**Comment by octonawish-akcodes** [comment]
I see that the gatekeeper has to provide `artifacthub-pkg.yaml` in each file which is a very lengthy process IMO to add it to every file, Is there any solution recommendation to bypass this?

**Comment by tegioz** [comment]
Please see point 2 [in this comment](https://github.com/artifacthub/hub/issues/2653#issuecomment-1383667937).
